editor/src/components/editor/store/reselect-examples.spec.tsx

```diff
@@ -0,0 +1,196 @@
+import { createSelector } from 'reselect'
+
+interface ExampleStore {
+  exNumber: number
+  exString: string
+}
+
+describe('Reselect Investigation', () => {
+  it('selectors are memoized', () => {
+    let numberOfTimesCalled = 0
+    const selector = createSelector(
+      (store: ExampleStore) => store.exNumber,
+      (store: ExampleStore) => store.exString,
+      (n, s) => {
+        numberOfTimesCalled += 1
+      },
+    )
+
+    const store: ExampleStore = {
+      exNumber: 5,
+      exString: 'hello',
+    }
+
+    selector(store)
+    expect(numberOfTimesCalled).toBe(1)
+
+    // calling it two more times will not increase numberOfTimesCalled!
+    selector(store)
+    selector(store)
+
+    expect(numberOfTimesCalled).toBe(1)
+  })
+
+  it('two instances of the same selector are memoized separately', () => {
+    let numberOfTimesCalled = { s1: 0, s2: 0 }
+    const selectorFn = (selectorKey: 's1' | 's2') =>
+      createSelector(
+        (store: ExampleStore) => store.exNumber,
+        (store: ExampleStore) => store.exString,
+        (n, s) => {
+          numberOfTimesCalled[selectorKey] += 1
+        },
+      )
+
+    const selector1 = selectorFn('s1')
+    const selector2 = selectorFn('s2')
+
+    const store: ExampleStore = {
+      exNumber: 5,
+      exString: 'hello',
+    }
+
+    selector1(store)
+    expect(numberOfTimesCalled.s1).toBe(1)
+    expect(numberOfTimesCalled.s2).toBe(0)
+    selector2(store)
+    expect(numberOfTimesCalled.s1).toBe(1)
+    expect(numberOfTimesCalled.s2).toBe(1)
+
+    // calling it two more times will not increase numberOfTimesCalled!
+    selector1(store)
+    selector1(store)
+    selector2(store)
+    selector2(store)
+
+    expect(numberOfTimesCalled.s1).toBe(1)
+    expect(numberOfTimesCalled.s2).toBe(1)
+  })
+
+  it('!!! WARNING: accidentally creating a new will not be memoized properly!!!!!', () => {
```

Comment:
```suggestion
  it('!!! WARNING: accidentally creating a new selector will not be memoized properly!!!!!', () => {
```

Also, It probably makes sense to move this to the very start of the test to increase visibility (though I'm sceptical that many will visit this file very often anyway, so this warning would probably make the most sense inside a comment in a share selectors file)